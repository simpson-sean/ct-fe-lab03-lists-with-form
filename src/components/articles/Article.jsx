import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => (
    <figure>
        <h1>{title}</h1>
        <h3>{author}</h3>
        <p>{description}</p>
    </figure>
)

Article.propTypes = {
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
};

export default Article;
