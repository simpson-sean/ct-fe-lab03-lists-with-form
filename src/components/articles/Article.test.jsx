import React from 'react';
import { render, screen } from '@testing-library/react';
import Article from './Article';

describe('Article', () => {
    it('displays an article', () => {
      render(
        <Article title="I like grapes" author="Farmer Dan" description="Grape are great" />
      );
  
      const article = screen.getByText('I like grapes','Farmer Dan', 'Grapes are great');
      expect(article).toMatchSnapshot();
    });
  });