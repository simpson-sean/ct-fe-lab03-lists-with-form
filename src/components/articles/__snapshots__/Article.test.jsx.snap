// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article displays an article 1`] = `
<h1>
  I like grapes
</h1>
`;
